src/app/model/ChessEngine.ts

```diff
@@ -17,6 +18,8 @@ export class ChessEngine implements ChessModel {
   readonly squaresWithPiece: SquareWithPiece;
   private promotingOnSquare: Square | undefined;
   private checkedKing: CheckedKing | undefined;
+  private castlingOnSquare: Square | undefined;
+  private gameRecord: GameRecord[] = [];
```

Comment:
Po co nam te nowy typ :) ? Nie wystarczy jakieś
`private movesHistory: PieceWasMoved[] = []` 
i w onPieceWasMoved dodawac zawsze do listy :) ? 
Mówiłem, że to jest proste w obecnej implementacji. 
Trochę chyba za dużo się napracowałaś.

Reply:
W sumie racja. Zamysł był trochę inny, ale nie wyszło tak jak, myślałam na początku i zostało coś takiego :P

Reply:
done